Port `@uppy/remote-sources` docs (#58)
transloadit/uppy.io: docs/presets/remote-sources.mdx

---
sidebar_position: 1
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

import UppyCdnExample from '/src/components/UppyCdnExample';

# Remote sources

> `@uppy/remote-sources` is a preset plugin to add all the available remote

Reviewer: ```suggestion
`@uppy/remote-sources` is a preset plugin (meaning it bundles and sets up other plugins) to add all the available remote
```

Reviewer: Although that is explained right after.

sources, such Instagram, Google Drive, Dropbox, and others to Uppy Dashboard in
one package.

:::note

Remote Sources requires Dashboard and automatically installs all its plugins to
it.

:::

## When should I use it?

`@uppy/remote-sources` aims to simplify the setup for adding Companion plugins,

> when you want to share the configuration between plugins. If you want your user

Reviewer: ```suggestion
when you want to share the configuration between plugins. If you want your users
```

> to upload files from any remote source, this plugin is recommended.

Reviewer: ```suggestion
to upload files from any of the remote sources that Uppy offers, this plugin is recommended.
```

> A [Companion](/docs/companion) instance is required for the Facebook plugin to

Reviewer: ```suggestion
A [Companion](/docs/companion) instance is required for the Remote Sources plugin to
```

> work. Companion handles authentication with Facebook, downloads the files, and

Reviewer: ```suggestion
work. Companion handles authentication with Remote Sources, such as Facebook, Dropbox, etc, downloads the files, and
```

Reviewer: Add link to Companion docs? Probably also to the Dashboard above.

uploads them to the destination. This saves the user bandwidth, especially
helpful if they are on a mobile connection.

You can self-host Companion or get a hosted version with any
[Transloadit plan](https://transloadit.com/pricing/).

## Install

> This plugin is published as the `@uppy/remote-sources` package on the npm

Reviewer: ```suggestion
```

Reviewer: We don't use that intro anywhere else in this section, I think.
